stable_audio_tools/models/conditioners.py

```diff
@@ -346,25 +359,22 @@ def forward(self, texts: tp.List[str], device: tp.Union[torch.device, str]) -> t
         )
 
         input_ids = encoded["input_ids"].to(device)
-        attention_mask = encoded["attention_mask"].to(device).to(torch.bool)
+        attention_mask = encoded["attention_mask"].to(device).bool()
 
         self.model.eval()
             
-        with torch.cuda.amp.autocast(dtype=torch.float16) and torch.set_grad_enabled(self.enable_grad):
+        with torch.cuda.amp.autocast(enabled=self.enable_grad, dtype=torch.float16 if self.enable_grad else torch.float32):
```

Comment:
Why would the autocast change based on the grad being enabled? I haven't used it in a long time, but I believe enable_grad was about fine-tuning the T5 model, shouldn't be related to autocasting.